build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/transport/FindTransportClassesPlugin.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.internal.precommit.transport;
+
+import org.elasticsearch.gradle.internal.conventions.precommit.PrecommitPlugin;
+import org.elasticsearch.gradle.util.GradleUtils;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.dsl.DependencyHandler;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.TaskProvider;
+
+import java.util.Map;
+
+public class FindTransportClassesPlugin extends PrecommitPlugin {
+    @Override
+    public TaskProvider<? extends Task> createTask(Project project) {
+        TaskProvider<FindTransportClassesTask> transportTestExistTask = project.getTasks()
+            .register("transportTestExistCheck", FindTransportClassesTask.class);
+        project.getPlugins().withType(JavaPlugin.class, javaPlugin -> {
```

Comment:
perhaps because I apply this to all `JavaPlugin` this task wis becoming a dependency to things like JarHell?